src/utilities/openlab-helper.ts

```diff
@@ -0,0 +1,83 @@
+import axios from 'axios';
+import configurations from '../configurations';
+import * as _ from 'lodash';
+import * as Joi from '@hapi/joi';
+import 'joi-extract-type';
+import { isAxiosError } from './axios-helper';
+import logger from './logger';
+import WrappedError from '../exceptions/wrapped-error';
+import { OpenLabRedirectInfo } from '../features/courses/course-types';
+import * as qs from 'qs';
+import formHelper, { unmergeStrategies } from './form-helper';
+import RederlyError from '../exceptions/rederly-error';
+
+/* eslint-disable @typescript-eslint/camelcase */
+export const openlabResponseValidationScheme = Joi.string().required();
+/* eslint-enable @typescript-eslint/camelcase */
+export type OpenLabResponse = Joi.extractType<typeof openlabResponseValidationScheme>;
+
+class OpenLabHelper {
+    openLabAxios = axios.create({
+        baseURL: configurations.openlab.url,
+        responseType: 'text',
+        timeout: configurations.openlab.requestTimeout,
+    });
+
+    OPENLAB_QS = qs.stringify({ webwork: 1 });
+
+    /**
+     * This function calls OpenLab with sensitive data and returns the js redirect script
+     * @param {object} OpenLabRedirectInfo generated by courseController.prepareOpenLabRedirect()
+     * @returns {string} a js script that will redirect the user to OpenLab
+     */
+    async askForHelp({
+        problem,
+        problemSetId,
+        courseId,
+        problemPath,
+        email,
+        studentName,
+        emailURL,
+        rawHTML,
+    }: OpenLabRedirectInfo): Promise<OpenLabResponse> {
+        let pgObject = '';
+        if (!_.isNil(rawHTML)) {
+            pgObject = Buffer.from(rawHTML).toString('base64');
+        } else {
+            // die -- we must have the renderedHTML for OpenLab
+            // and we don't have any anonymous ways of id-ing the users involved...
```

Comment:
We could pass in the student's id from session.

Reply:
Unless logging meta already does that for us.

Reply:
It does, I think this is perfect